src/bin/cargo/commands/add.rs

```diff
@@ -224,6 +230,7 @@ pub fn exec(gctx: &mut GlobalContext, args: &ArgMatches) -> CliResult {
 
 fn parse_dependencies(gctx: &GlobalContext, matches: &ArgMatches) -> CargoResult<Vec<DepOp>> {
     let path = matches.get_one::<String>("path");
+    let base = matches.get_one::<String>("base");
```

Comment:
I'm not seeing an unstable gate on this flag

Reply:
How do I do that check?

We changed `path-bases` to a Cargo feature, so there is no `-Z` flag. I also see that the `--public` flag is unstable, but has no checks for `-Z` flags or Cargo features.

Reply:
Well, thats a bug that `public` doesn't check...

In your case, the options are:
- Gate it on `-Zunstable-options`
- In `resolve_dependency`, you have a `Package` so you can get a `Manifest` and check `unstable_features` for this

Reply:
Ah, I didn't realize that `spec` was the `Package` for the current package. Thanks!

Reply:
Looks like this was resolved by delegating to `lookup_path_base`